easybuild/tools/filetools.py

```diff
@@ -1218,61 +1224,60 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None, use_git_am=Fa
         raise EasyBuildError("Can't patch directory %s: no such directory", dest)
 
     # use absolute paths
-    apatch = os.path.abspath(patch_file)
-    adest = os.path.abspath(dest)
+    abs_patch_file = os.path.abspath(patch_file)
+    abs_dest = os.path.abspath(dest)
 
     # Attempt extracting the patch if it ends in .patch.gz, .patch.bz2, .patch.xz
     # split in name + extension
-    apatch_root, apatch_file = os.path.split(apatch)
-    apatch_name, apatch_extension = os.path.splitext(apatch_file)
+    patch_filestem, patch_extension = os.path.splitext(os.path.split(abs_patch_file)[1])
```

Comment:
I would go with `patch_filename` (or `patch_name`), `filestem` is so close to `filesystem` that it's *very* confusing

Reply:
then I'd rather use `patch_stem`. "stem" is commonly used as the filename without the extension which is why I wanted to highlight it here. Or maybe `patch_name` but I'd like to avoid `filename` because that usually includes the extension

Reply:
Then I'd converge on `patch_name`, `patch_stem` is fine too.

Reply:
Ok, will go with `patch_stem` and a comment above it.

Note: I also removed the `change_dir` below: https://github.com/easybuilders/easybuild-framework/pull/3414/commits/9956c30917e0e20b6357b4c3cd905da4bdb1e6c7